test/e2e/deployment.go

```diff
@@ -149,6 +152,23 @@ func checkDeploymentRevision(c *clientset.Clientset, ns, deploymentName, revisio
 	return deployment, newRC
 }
 
+func stopDeployment(c *clientset.Clientset, oldC client.Interface, ns, deploymentName string) {
+	deployment, err := c.Extensions().Deployments(ns).Get(deploymentName)
+	Expect(err).NotTo(HaveOccurred())
+
+	Logf("stopping deployment %s", deploymentName)
+	reaper, err := kubectl.ReaperFor(extensions.Kind("Deployment"), oldC)
+	Expect(err).NotTo(HaveOccurred())
+	timeout := 1 * time.Minute
+	err = reaper.Stop(ns, deployment.Name, timeout, api.NewDeleteOptions(0))
+	Expect(err).NotTo(HaveOccurred())
+
+	Logf("ensuring deployment %s was stopped", deploymentName)
+	_, err = c.Extensions().Deployments(ns).Get(deployment.Name)
+	Expect(err).To(HaveOccurred())
+	Expect(errors.IsNotFound(err)).To(BeTrue())
```

Comment:
I'll add just checking rcs, pods have grace period, which I'd have to wait, not sure if it's worth.